cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryConfig.java

```diff
@@ -0,0 +1,166 @@
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Represents the repository configuration of a namespace. This class needs to be GSON serializable.
+ */
+public class RepositoryConfig {
+  private final Provider provider;
+  private final String link;
+  private final String defaultBranch;
+  private final String pathPrefix;
+  private final AuthConfig auth;
+
+  RepositoryConfig(Provider provider, String link, String defaultBranch, AuthConfig authConfig,
+                   @Nullable String pathPrefix) {
+    this.provider = provider;
+    this.link = link;
+    this.defaultBranch = defaultBranch;
+    this.auth = authConfig;
+    this.pathPrefix = pathPrefix;
+  }
+
+  public Provider getProvider() {
+    return provider;
+  }
+
+  public String getLink() {
+    return link;
+  }
+
+  public String getDefaultBranch() {
+    return defaultBranch;
+  }
+  
+  @Nullable
+  public String getPathPrefix() {
+    return pathPrefix;
+  }
+
+  public AuthConfig getAuth() {
+    return auth;
+  }
+
+  public boolean isValid() {
+    return provider != null && link != null && defaultBranch != null && auth != null && auth.isValid();
+  }
+
+  /**
+   * Builder used to build {@link RepositoryConfig}
+   */
+  public static final class Builder {
+    private Provider provider;
+    private String link;
+    private String defaultBranch;
+    private String pathPrefix;
+    private AuthType authType;
+    private String tokenName;
+    private String username;
+
+    public Builder() {
+      // no-op
+    }
+
+    public Builder(RepositoryConfig repoConfig) {
+      this.provider = repoConfig.getProvider();
+      this.link = repoConfig.getLink();
+      this.defaultBranch = repoConfig.getDefaultBranch();
+      this.pathPrefix = repoConfig.getPathPrefix();
+      if (repoConfig.getAuth() != null) {
+        this.authType = repoConfig.getAuth().getType();
+        this.tokenName = repoConfig.getAuth().getTokenName();
+        this.username = repoConfig.getAuth().getUsername();
+      }
+    }
+
+    public Builder setProvider(Provider provider) {
+      this.provider = provider;
+      return this;
+    }
+
+    public Builder setLink(String link) {
+      this.link = link;
+      return this;
+    }
+
+    public Builder setDefaultBranch(String defaultBranch) {
+      this.defaultBranch = defaultBranch;
+      return this;
+    }
+
+    public Builder setPathPrefix(String pathPrefix) {
+      this.pathPrefix = pathPrefix;
+      return this;
+    }
+
+    public Builder setAuthType(AuthType authType) {
+      this.authType = authType;
+      return this;
+    }
+
+    public Builder setTokenName(String tokenName) {
+      this.tokenName = tokenName;
+      return this;
+    }
+    
+    public Builder setUsername(String username) {
+      this.username = username;
+      return this;
+    }
+
+    public RepositoryConfig build() {
+      return new RepositoryConfig(provider, link, defaultBranch,
```

Comment:
should call isValid() before returning it. We don't want to ever allow creating an invalid RepositoryConfig in our java code

Reply:
added the validate method and throw list of errors once validation fails